--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -104,7 +104,7 @@ if (BUILD_FOR_DESKTOP)
     BUILD_IN_SOURCE 1
     CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}"
     #PATCH_COMMAND patch -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/003-linenoise-oflag.patch
-    BUILD_COMMAND gcc -c -Wall -Werror -fpic linenoise.c COMMAND gcc --shared -o liblinenoise.so linenoise.o)
+    BUILD_COMMAND ${CMAKE_C_COMPILER} -c -Wall -Werror -fpic linenoise.c COMMAND ${CMAKE_C_COMPILER} --shared -o liblinenoise.so linenoise.o)
 
   include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}